qa/workunits/suites/mgr_mod_snap_schedule.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/bash
+TESTDIR=$(dirname $0)
+# run for 16 minutes
+RUN_FOR=16
+EXPECTED_SNAPS=15
+RNG_SEED=${RNG_SEED:-$(date +%s)}
+
+echo "TESTDIR=$TESTDIR"
+echo "RUN_FOR=$RUN_FOR"
+echo "EXPECTED_SNAPS=$EXPECTED_SNAPS"
+echo "RNG_SEED=$RNG_SEED"
+echo "PYTHON=$(which python3)"
+
+ceph mgr module enable snap_schedule
+sleep 2
+ceph config set mgr mgr/snap_schedule/allow_m_granularity True
+sleep 2
+mkdir z
+pushd z
+fs_path=$(pwd | sed -e "s|^$CEPH_MNT||g")
+ceph fs snap-schedule add $fs_path 1M
+
+(python3 -c "import os
+import random
+import time
+from datetime import datetime
+
+random.seed($RNG_SEED)
+
+fd = os.open('./data-file', os.O_WRONLY | os.O_CREAT, 0o777)
+assert fd is not None
+
+start = datetime.now()
+data = 'A'
+while True:
+    end = datetime.now()
+    elapsed = end - start
+    if elapsed.total_seconds() >= int($RUN_FOR * 60):
+        break
+
+    s = data * 1024
+    os.write(fd, s.encode('ascii'))
+    next = ord(data) - ord('A') + 1
+    next %= 26
+    data = chr(next + ord('A'))
+    time.sleep(random.randint(1, 13))
+
+os.close(fd)
+") &
+
+bg_pid=$!
+
+# initialize the Random Number Generator
+RANDOM=$RNG_SEED
+
+let -i start=$(date +%s)
+while true;
+do
+  let -i now=$(date +%s)
+  let -i elapsed=$((now - start))
+  if [ $elapsed -ge $(($RUN_FOR * 60)) ]; then
+    break
+  fi
+  ceph fs snap-schedule status $fs_path || echo "$0: snap-schedule status command failed with status $?"
```

Comment:
So this is a lot of run up to this one task that's the only regular interaction with the mgr (to evaluate thrashing). We're not even really evaluating whether the `snap_schedule` module recovers at all. We're not regularly modifying the snap schedule (despite the name of this script).

I think this should be dropped. Instead incorporate `qa/suites/fs/workload/tasks/3-snaps` into the thrash suite which is probably a good idea even without mgr thrashing.

You can modify 3-snaps/yes.yaml to add a background_exec job which queries the snap schedule status regularly.